clients/dataset/dataset.go

```diff
@@ -0,0 +1,78 @@
+package dataset
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// ErrInvalidDatasetAPIResponse is returned when the dataset api does not respond
+// with a valid status
+type ErrInvalidDatasetAPIResponse struct {
+	expectedCode int
+	actualCode   int
+	uri          string
+}
+
+func (e ErrInvalidDatasetAPIResponse) Error() string {
+	return fmt.Sprintf("invalid response from dataset api - should be: %d, got: %d, path: %s",
+		e.expectedCode,
+		e.actualCode,
+		e.uri,
+	)
+}
+
+var _ error = ErrInvalidDatasetAPIResponse{}
+
+// Client is a dataset api client which can be used to make requests to the server
+type Client struct {
+	cli *http.Client
+	url string
+}
+
+// New creates a new instance of Client with a given filter api url
+func New(datasetAPIURL string) *Client {
+	return &Client{
+		cli: &http.Client{Timeout: 5 * time.Second},
+		url: datasetAPIURL,
+	}
+}
+
+// Healthcheck calls the healthcheck endpoint on the api and alerts the caller of any errors
+func (c *Client) Healthcheck() (string, error) {
+	resp, err := c.cli.Get(c.url + "/healthcheck")
+	if err != nil {
+		return "dataset-api", err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "dataset-api", &ErrInvalidDatasetAPIResponse{http.StatusOK, resp.StatusCode, "/healthcheck"}
+	}
+
+	return "", nil
+}
+
+// GetDataset ...
+func (c *Client) GetDataset(id, edition, version string) (m Model, err error) {
```

Comment:
Probably shouldn't call it GetDataset when it's actually fetching the Version - could cause confusion when the aim is to actually retrieve dataset metadata